pkg/manager/internal.go

```diff
@@ -243,9 +236,6 @@ func (cm *controllerManager) SetFields(i interface{}) error {
 	if _, err := inject.InjectorInto(cm.SetFields, i); err != nil {
 		return err
 	}
-	if _, err := inject.StopChannelInto(cm.internalStop, i); err != nil {
```

Comment:
Shoulnd't this get replaced with an `inject.ContextInto`?

Reply:
It depends, given that we should really pass context as the first parameter, instead of injecting it into structs

Reply:
Well the problem is that due to this terrible inject pattern the code will continue to compile but not work as before (I would assume it will just panic now for most ppl that relied on this). Can we maybe keep this for now, inject `cm.internalCtx.Done` and print a big fat deprecation warning in `inject.StopChannelInto` if this happens?

How is kubebuilder the CLI going to migrate projects?

Reply:
I see, would we pass the ctx.Done() channel here then?

Reply:
> Can we maybe keep this for now, inject cm.internalCtx.Done and print a big fat deprecation warning in inject.StopChannelInto if this happens?

:+1: 

> How is kubebuilder the CLI going to migrate projects?

Sounds like we'll keep this interface and injection in for now, so no changes necessary on the kubebuilder scaffolding side related to this particular aspect of the PR?

Reply:
@alvaroaleman Mind reviewing the last changes related to `StopChannelInto`, I'm a little confused about this interface and the whole injection code left me with some questions, but we can fix those later. For the purpose of this PR, can you double check that the logic here is restored as you'd expect?

Reply:
I don't see anything related to that, it still seems to be gone, am I completely missing it?

Reply:
Done, must have forgotten to push the changes

Reply:
I still only see the removed code?

Reply:
* actually pushed now *